ambari-server/src/main/java/org/apache/ambari/server/controller/internal/HostComponentResourceProvider.java

```diff
@@ -396,6 +396,7 @@ public RequestStatusResponse install(String cluster, String hostname, Collection
     Map<String, String> requestInfo = new HashMap<>();
     requestInfo.put("context", String.format("Install components on host %s", hostname));
     requestInfo.put(CLUSTER_PHASE_PROPERTY, CLUSTER_PHASE_INITIAL_INSTALL);
+    requestInfo.putAll(RequestOperationLevel.propertiesFor(Resource.Type.HostComponent, cluster));
```

Comment:
As this was a really difficult problem to understand, maybe add a comment here about maint mode describing why this is necessary.